src/Runner.Listener/SelfUpdater.cs

```diff
@@ -131,6 +131,11 @@ public async Task<bool> SelfUpdate(AgentRefreshMessage updateMessage, IJobDispat
                 // For L0, we will skip execute update script.
                 if (string.IsNullOrEmpty(Environment.GetEnvironmentVariable("_GITHUB_ACTION_EXECUTE_UPDATE_SCRIPT")))
                 {
+                    string flagFile = "update.finished";
+                    if (File.Exists(flagFile))
```

Comment:
`IOUtil.Delete` should be a little safer and you don't have to check with `File.Exists`